src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Extensions/ITrainingProgrammeExtensionsTest/WhenDetermingWhetherACourseIsActive.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.EmployerCommitments.Application.Extensions;
+using SFA.DAS.EmployerCommitments.Domain.Models.ApprenticeshipCourse;
+
+namespace SFA.DAS.EmployerCommitments.Application.UnitTests.Extensions.ITrainingProgrammeExtensionsTest
+{
+    [TestFixture]
+    public class WhenDetermingWhetherACourseIsActive
+    {
+        [TestCase("2016-01-01", "2016-12-01", "2016-06-01", true, Description = "Within date range")]
+        [TestCase("2016-01-15", "2016-12-15", "2016-01-01", true, Description = "Within date range - ignoring start day")]
+        [TestCase("2016-01-15", "2016-12-15", "2016-12-30", true, Description = "Within date range - ignoring end day")]
+        [TestCase(null, "2016-12-01", "2016-06-01", true, Description="Within date range with no defined course start date")]
+        [TestCase("2016-01-01", null, "2016-06-01", true, Description = "Withing date range, with no defined course end date")]
+        [TestCase(null, null, "2016-06-01", true, Description = "Within date range, with no defined course effective dates")]
+        [TestCase("2016-01-01", "2016-12-01", "2015-06-01", false, Description = "Outside (before) date range")]
+        [TestCase("2016-01-01", "2016-12-01", "2017-06-01", false, Description = "Outside (after) date range")]
```

Comment:
I'd be temped to add test cases for either side of both boundaries, i.e.

[TestCase("2016-01-01", "2016-12-01", "2016-01-01", true, Description = "Within date range (first day)")]
[TestCase("2016-01-01", "2016-12-01", "2016-12-31", true, Description = "Within date range (last day)")]
[TestCase("2016-01-01", "2016-12-01", "2015-12-31", false, Description = "Outside (just before) date range")]
[TestCase("2016-01-01", "2016-12-01", "2017-01-01", false, Description = "Outside (just after) date range")]